packages/shylu/shylu_dd/frosch/src/Adapters/Thyra_FROSchFactory_def.hpp

```diff
@@ -130,22 +130,77 @@ namespace Thyra {
         FROSCH_ASSERT(paramList_->isParameter("FROSch Preconditioner Type"),"FROSch Preconditioner Type is not defined!");
 
         if (!paramList_->get("FROSch Preconditioner Type","TwoLevelPreconditioner").compare("AlgebraicOverlappingPreconditioner")) {
+            // Extract the repeated map
             ConstMapPtr repeatedMap = Teuchos::null;
             if (paramList_->isParameter("Repeated Map")) {
-                repeatedMap = ExtractPtrFromParameterList<ConstMap>(*paramList_,"Repeated Map");
+                repeatedMap = ExtractPtrFromParameterList<Map>(*paramList_,"Repeated Map").getConst();
+                if (repeatedMap.is_null()) {
+                    if (A->getMap()->lib()==Xpetra::UseTpetra) { // If coordinatesList.is_null(), we look for Tpetra/Epetra RCPs
+                        Teuchos::RCP<const Tpetra::Map<LO,GO,NO> > repeatedMapTmp = ExtractPtrFromParameterList<const Tpetra::Map<LO,GO,NO> >(*paramList_,"Repeated Map");
```

Comment:
Use`auto` for the left-hand side, since the right-hand side already determines the type.

Reply:
This way I don't have to look up the type in a different file. I prefer to explicitly write down the types.